src/Documentation/Getting-Started-With-Orleans/Grain-LifeCycle.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: page
+title: Grain LifeCycle
+---
+
+
+## Grain LifeCycle
+
+We say grains virtually always exist but here we want to describe different stages of a grain's life.
+A grain is like a class definition, it is nothing unless it is on the memory and can execute code.
+An instance of a class is instantiated with the `new` keyword in C#.
+an instance of a grain is called an activation.
+The grain's instances are differentiated by their key and a grain with a specific key is either activated or not.
+You don't control this by using construction/destruction keywords and instead Orleans runtime manages this for you.
+
+All possible grains in the whole key space virtually always exist
+and it means whenever you call a method of a grain, if it is not active, it will get activated by the runtime.
+Also after the grain passes a certain amount of time in idle state, the runtime will deactivate the grain (i.e. destroys it).
+A grain is in idle state if it is not currently processing a message and its cueue of messages is empty.
+
+## callbacks 
+
+Just like the class which can have constructors and (destructors in some languages), grains have
+`OnActivateAsync` and `OnDeactivateAsync` virtual methods which you can override.
+
+## Different grain types and their differences regarding life cycle
+
+Your grains can be persistent and store their state in storage
+or have no state which is stored in storage. 
+In case of the later, all activations are created equal and there is no difference between them.
```

Comment:
What you mean by that? Even without state a grain is differentiated by its ID/Key, not the state.